src/main/java/nl/novi/behuiswerkles11/controllers/TelevisionsController.java

```diff
@@ -1,39 +1,71 @@
 package nl.novi.behuiswerkles11.controllers;
 
+import jakarta.validation.Valid;
+import nl.novi.behuiswerkles11.dtos.TelevisionRequestDto;
+import nl.novi.behuiswerkles11.dtos.TelevisionDto;
+import nl.novi.behuiswerkles11.mappers.TelevisionModelMapper;
+import nl.novi.behuiswerkles11.models.Television;
+import nl.novi.behuiswerkles11.services.TelevisionService;
 import org.springframework.http.HttpStatus;
-import org.springframework.http.HttpStatusCode;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
+import java.util.List;
+
 @RestController
 @RequestMapping("televisions")
 public class TelevisionsController {
+    private final TelevisionService televisionService;
+    private final TelevisionModelMapper televisionModelMapper;
+
+    public TelevisionsController(TelevisionService televisionService, TelevisionModelMapper televisionModelMapper) {
+        this.televisionService = televisionService;
+        this.televisionModelMapper = televisionModelMapper;
+    }
+
     @GetMapping()
-    public ResponseEntity<String> getTelevisions() {
-//        return ResponseEntity.ok("televisions");
-        return new ResponseEntity<>("all televisions", HttpStatus.OK);
+    public ResponseEntity<List<TelevisionDto>> getTelevisions() {
+        List<Television> televisions = televisionService.getTelevisions();
+        List<TelevisionDto> televisionDtos = televisionModelMapper.translateToDtos(televisions);
+
+        return ResponseEntity.ok(televisionDtos);
     }
 
+    //    Request mapping voor het ophalen van 1 tv op basis van het id
     @GetMapping("/{id}")
-    public ResponseEntity<String> getOneTelevision(@PathVariable Integer id) {
-        return ResponseEntity.ok("televisionId: " + id);
+    public ResponseEntity<Television> getTelevision(@PathVariable Long id) {
+        return ResponseEntity.ok(televisionService.getTelevision(id));
     }
 
     // Omdat we hier iets posten in plaats van aanvragen, hebben we geen body in onze return. Vandaar dat er <void> staat na het definieren van de ResponseEntity.
+//    Request mapping voor het toevoegen van 1 tv
     @PostMapping()
-    public ResponseEntity<Void> addTelevision(@RequestBody String television) {
-        System.out.println(television);
-        return ResponseEntity.created(null).build();
+    public ResponseEntity<TelevisionDto> addTelevision(@Valid @RequestBody TelevisionRequestDto television) {
+        Television savedTelevision = televisionService.addTelevision(televisionModelMapper.translateToTelevision(television));
+        return ResponseEntity.status(HttpStatus.CREATED).body(televisionModelMapper.translateToDto(savedTelevision));
```

Comment:
Een body is niet verplicht voor een POST mapping, wel handig soms. Wat wel nodig is, is de location header. Dat is de URL voor het GET request om de zojuist gemaakte entity weer op te halen (dus "/televisions/1" o.i.d.). De code hiervoor heb ik gedeeld is gedeeld in de huiswerkles en in de les.